github_cog.py

```diff
@@ -0,0 +1,111 @@
+import discord
+import github.GithubException
+from discord.ext import commands
+from github import Github
+from contextlib import closing
+import studybot
+
+
+# noinspection PyBroadException
+class GitHub_Integration(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command(name='github-join', help=studybot.help_dict.get('github-join'))
+    async def join(self, ctx, github_name='null'):
+        # Make sure they included the github_name arg, and send a response.
+        if github_name == 'null':
+            msg = 'You need to include your GitHub Username like this: '
+            msg += '`!github-join StudyBot`'
+            await ctx.message.reply(msg)
+            return
+        else:
+            g = Github(studybot.TOKENS.get('GITHUB_API_KEY'))
+            # Get the user object.
+            try:
+                g_user = g.get_user(github_name)
+            except:
+                await ctx.message.reply('The GitHub user you requested cannot be found. Double-Check your spelling.')
+                return()
+            # Get the organization object.
+            try:
+                g_org = g.get_organization(studybot.github_org_name)
+            except:
+                msg = 'The GitHub organization cannot be found. Please try again later, '
+                msg += 'or contact a mod if the issue persists.'
+                await ctx.message.reply(msg)
+                return()
+
+            # Verify that the user has not already invited a GitHub account to the org.
+            with closing(studybot.db_conn.cursor()) as conn:
+                async with studybot.lock:
+                    query = 'SELECT github_id FROM github_invites WHERE discord_id = ?'
+                    conn.execute(query, (ctx.author.id, ))
+                    value = conn.fetchone()
+                    if value is None:
+                        # Verify that the user is not already in the org.
+                        if g_org.has_in_members(g_user):
+                            await ctx.message.reply('That user is already in the organization, and cannot be invited.')
+                            return()
+                        # Invite the user to the organization.
+                        else:
+                            g_org.invite_user(g_user)
```

Comment:
Depending on the order stuff failes in a user could invite more than one user if the db add fails after this line ran.

Reply:
I'd rather have a user be able to add a second person than have it block the user from adding anyone if the API changes and their invite fails.

Reply:
ok